packages/core/src/modules/connections/ConnectionsModule.ts

```diff
@@ -197,6 +198,91 @@ export class ConnectionsModule {
     return this.connectionService.getAll()
   }
 
+  /**
+   * Allows for the addition of connectionType to the record.
+   *  Either updates or creates an array of string conection types
+   * @param connectionId
+   * @param type
+   * @throws AriesFrameworkError if the record corrisponding to the provided connectionId cannot be found
+   */
+  public async addConnectionType(connectionId: string, type: ConnectionType | string) {
+    const record = await this.findById(connectionId)
```

Comment:
To optimize a bit the code, I think you can call `await this.getById(connectionId)`, which already verifies if the connection exists and throw exception if not found.